python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -747,7 +732,18 @@ def write(self, value: Any, timeout: Optional[float] = None) -> None:
 
     def read(self, timeout: Optional[float] = None) -> Any:
         self.ensure_registered_as_reader()
-        actor_id = self._get_self_actor_id()
+        actor_id = ray.get_runtime_context().get_actor_id()
+        # if actor_id is None, read was called by the driver
+        # if the driver is an actor, driver_actor_id will be set to that actor id
+        if actor_id is None or actor_id == self._driver_actor_id:
```

Comment:
want to highlight here that just `if actor_id not in self._channel_dict` would probably also work here, but we came to the conclusion that it would lead to some misleading, more difficult to debug execution in the case that an incorrect actor was actually trying to read the channel